<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CustomBioInfoFunctionsHumanGenome">
<title>CustomBioInfoFunctionsHumanGenome • CustomBioInfoFunctionsHumanGenome</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CustomBioInfoFunctionsHumanGenome">
<meta property="og:description" content="CustomBioInfoFunctionsHumanGenome">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CustomBioInfoFunctionsHumanGenome</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/CustomBioInfoFunctionsHumanGenome.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/vinodsinghjnu/CustomBioInfoFunctionsHumanGenome/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>CustomBioInfoFunctionsHumanGenome</h1>
                        <h4 data-toc-skip class="author">Vinod
Singh</h4>
            
            <h4 data-toc-skip class="date">2024-04-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vinodsinghjnu/CustomBioInfoFunctionsHumanGenome/blob/HEAD/vignettes/CustomBioInfoFunctionsHumanGenome.Rmd" class="external-link"><code>vignettes/CustomBioInfoFunctionsHumanGenome.Rmd</code></a></small>
      <div class="d-none name"><code>CustomBioInfoFunctionsHumanGenome.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/IRanges" class="external-link">IRanges</a></span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span>; <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vinodsinghjnu/CustomBioInfoFunctionsHumanGenome" class="external-link">CustomBioInfoFunctionsHumanGenome</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">1. Introduction:<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This Package contains some customized functions for common operations
used in human genome analysis.</p>
</div>
<div class="section level2">
<h2 id="availability-and-installation">2. Availability and Installation<a class="anchor" aria-label="anchor" href="#availability-and-installation"></a>
</h2>
<p>The development version of
<code>CustomBioInfoFunctionsHumanGenome</code> package is available at
<a href="https://github.com/vinodsinghjnu/CustomBioInfoFunctionsHumanGenome" class="external-link uri">https://github.com/vinodsinghjnu/CustomBioInfoFunctionsHumanGenome</a>
and can be installed as</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"vinodsinghjnu/CustomBioInfoFunctionsHumanGenome"</span>,build_vignettes <span class="op">=</span> <span class="cn">TRUE</span> <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="functions">3. Functions<a class="anchor" aria-label="anchor" href="#functions"></a>
</h2>
<div class="section level3">
<h3 id="addgrinformation">3.1 addGrInformation<a class="anchor" aria-label="anchor" href="#addgrinformation"></a>
</h3>
<div class="section level4">
<h4 id="description">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description"></a>
</h4>
<p>add assembly information to the genomic range.</p>
</div>
<div class="section level4">
<h4 id="usage">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage"></a>
</h4>
<p><code>addGrInformation(gr.f=gr, assmblyName='hg19')</code></p>
</div>
<div class="section level4">
<h4 id="arguments">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments"></a>
</h4>
<ul>
<li>
<code>gr.f</code>: A genomic range.</li>
<li>
<code>assmblyName</code>: human genome assembly name i.e., hg19 or
hg38 or t2t. Default is <code>hg19</code>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="details">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details"></a>
</h4>
<p>This function will add genomic length and assembly name to given
genomic ranges (Human genome only). It will remove the non-standard
chromosomes from genomic ranges and report the bad genomic ranges for
the selected genome assembly.</p>
</div>
<div class="section level4">
<h4 id="value">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value"></a>
</h4>
<p>returns the input genomic range along with assembly information.</p>
</div>
<div class="section level4">
<h4 id="examples">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span>,  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">110</span>,<span class="fl">105</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">120</span>, <span class="fl">150</span> <span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">outGr</span><span class="op">=</span><span class="fu"><a href="../reference/addGrInformation.html">addGrInformation</a></span><span class="op">(</span>gr.f<span class="op">=</span><span class="va">gr</span>, assmblyName<span class="op">=</span><span class="st">'hg19'</span><span class="op">)</span></span>
<span><span class="va">outGr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 3 ranges and 0 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chr1     1-100      +</span></span>
<span><span class="co">##   [2]     chr1   110-120      -</span></span>
<span><span class="co">##   [3]     chr1   105-150      +</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from 2 genomes (hg19, NA)</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="pctoverlap_of_firstgrtosecondgr">3.2 pctOverlap_Of_FirstGrToSecondGr<a class="anchor" aria-label="anchor" href="#pctoverlap_of_firstgrtosecondgr"></a>
</h3>
<div class="section level4">
<h4 id="description-1">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-1"></a>
</h4>
<p>Percent overlap of a GenomicRange with other</p>
</div>
<div class="section level4">
<h4 id="usage-1">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-1"></a>
</h4>
<p><code>pctOverlap_Of_FirstGrToSecondGr(FirstContext=gr1, SecondContext=gr2)</code></p>
</div>
<div class="section level4">
<h4 id="arguments-1">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-1"></a>
</h4>
<ul>
<li>
<code>FirstContext</code>: GenomicRange object. (query:
overlapped)</li>
<li>
<code>SecondContext</code>: GenomicRange object. (overlapped
to)</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-1">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-1"></a>
</h4>
<p>Percent overlap of a genomic range with other.</p>
</div>
<div class="section level4">
<h4 id="value-1">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-1"></a>
</h4>
<p>Percent overlap of first genomic range to second genomic range</p>
</div>
<div class="section level4">
<h4 id="examples-1">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-1"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span>,  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">110</span>,<span class="fl">105</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">120</span>, <span class="fl">150</span> <span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span>, <span class="st">"+"</span><span class="op">)</span>,  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">12</span>,<span class="fl">60</span>, <span class="fl">105</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">25</span>, <span class="fl">50</span>, <span class="fl">70</span>, <span class="fl">115</span> <span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/pctOverlap_Of_FirstGrToSecondGr.html">pctOverlap_Of_FirstGrToSecondGr</a></span><span class="op">(</span>FirstContext<span class="op">=</span><span class="va">gr1</span>, SecondContext<span class="op">=</span><span class="va">gr2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 45.85987</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="emptychrgranges">3.3 emptyChrGranges<a class="anchor" aria-label="anchor" href="#emptychrgranges"></a>
</h3>
<div class="section level4">
<h4 id="description-2">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-2"></a>
</h4>
<p>Create empty chromosome GenomicRange object for a given human genome
assembly</p>
</div>
<div class="section level4">
<h4 id="usage-2">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-2"></a>
</h4>
<p><code>emptyChrGranges(assmblyName='hg19'))</code></p>
</div>
<div class="section level4">
<h4 id="arguments-2">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-2"></a>
</h4>
<ul>
<li>
<code>assmblyName</code>: hg19 or hg38 or t2t</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-2">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-2"></a>
</h4>
<p>Create empty chromosome GenomicRange object for a given human genome
assembly</p>
</div>
<div class="section level4">
<h4 id="value-2">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-2"></a>
</h4>
<p>Create empty chromosome GenomicRange object for a given human genome
assembly</p>
<p>Note: Output object is labelled with assembly information.</p>
</div>
<div class="section level4">
<h4 id="examples-2">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-2"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hg_19_Chr.gr</span><span class="op">=</span><span class="fu"><a href="../reference/emptyChrGranges.html">emptyChrGranges</a></span><span class="op">(</span><span class="st">'hg19'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hg_19_Chr.gr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 6 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames      ranges strand |                    Seqs</span></span>
<span><span class="co">##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |          &lt;DNAStringSet&gt;</span></span>
<span><span class="co">##   [1]     chr1 1-249250621      * | NNNNNNNNNN...NNNNNNNNNN</span></span>
<span><span class="co">##   [2]     chr2 1-243199373      * | NNNNNNNNNN...NNNNNNNNNN</span></span>
<span><span class="co">##   [3]     chr3 1-198022430      * | NNNNNNNNNN...NNNNNNNNNN</span></span>
<span><span class="co">##   [4]     chr4 1-191154276      * | NNNNNNNNNN...NNNNNNNNNN</span></span>
<span><span class="co">##   [5]     chr5 1-180915260      * | NNNNNNNNNN...NNNNNNNNNN</span></span>
<span><span class="co">##   [6]     chr6 1-171115067      * | NNNNNNNNNN...NNNNNNNNNN</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from hg19 genome</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="getgbins">3.4 getGbins<a class="anchor" aria-label="anchor" href="#getgbins"></a>
</h3>
<div class="section level4">
<h4 id="description-3">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-3"></a>
</h4>
<p>Create GenomicRanges object of given the bin size for human
genome.</p>
</div>
<div class="section level4">
<h4 id="usage-3">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-3"></a>
</h4>
<p><code>getGbins(assmblyName='hg19', binSize=1000 )</code></p>
</div>
<div class="section level4">
<h4 id="arguments-3">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-3"></a>
</h4>
<ul>
<li>
<code>assmblyName</code>: hg19 or hg38 0r t2t</li>
<li>
<code>binSize</code>: size of the genomic block</li>
<li>
<code>chrName</code>: name of the chromosome, if not given then all
chromosomes are considered (UCSC format)</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-3">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-3"></a>
</h4>
<p>Create GenomicRanges object of given the bin size for human
genome.</p>
</div>
<div class="section level4">
<h4 id="value-3">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-3"></a>
</h4>
<p>GenomicRanges object of given the bin size</p>
<p>Note: Output object is labelled with assembly information.</p>
</div>
<div class="section level4">
<h4 id="examples-3">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-3"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hg_19_Bins.gr</span><span class="op">=</span><span class="fu"><a href="../reference/getGbins.html">getGbins</a></span><span class="op">(</span>assmblyName<span class="op">=</span><span class="st">'hg19'</span>, binSize<span class="op">=</span><span class="fl">1000</span> <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">hg_19_Bins.gr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 6 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames    ranges strand | CpG_counts</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |  &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]     chr1    1-1000      * |          0</span></span>
<span><span class="co">##   [2]     chr1 1001-2000      * |          0</span></span>
<span><span class="co">##   [3]     chr1 2001-3000      * |          0</span></span>
<span><span class="co">##   [4]     chr1 3001-4000      * |          0</span></span>
<span><span class="co">##   [5]     chr1 4001-5000      * |          0</span></span>
<span><span class="co">##   [6]     chr1 5001-6000      * |          0</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from hg19 genome</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="dnaseqsforpattern">3.5 DNASeqsForPattern<a class="anchor" aria-label="anchor" href="#dnaseqsforpattern"></a>
</h3>
<div class="section level4">
<h4 id="description-4">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-4"></a>
</h4>
<p>Generate all possible DNA sequences of a <a href="https://genomevolution.org/wiki/index.php/Ambiguous_nucleotide" class="external-link">Ambiguous
nucleotide sequence</a></p>
</div>
<div class="section level4">
<h4 id="usage-4">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-4"></a>
</h4>
<p><code>DNASeqsForPattern(pat='NYYN')</code></p>
</div>
<div class="section level4">
<h4 id="arguments-4">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-4"></a>
</h4>
<ul>
<li>
<code>pat</code>: a <a href="https://genomevolution.org/wiki/index.php/Ambiguous_nucleotide" class="external-link">ambiguous
nucleotide sequence</a>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-4">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-4"></a>
</h4>
<p>Generate all possible DNA sequences of a <a href="https://genomevolution.org/wiki/index.php/Ambiguous_nucleotide" class="external-link">Ambiguous
nucleotide sequence</a></p>
</div>
<div class="section level4">
<h4 id="value-4">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-4"></a>
</h4>
<p>A vector of all possible DNA sequences for a given Ambiguous
nucleotide sequence</p>
</div>
<div class="section level4">
<h4 id="examples-4">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-4"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DNA_seqs</span><span class="op">=</span><span class="fu"><a href="../reference/DNASeqsForPattern.html">DNASeqsForPattern</a></span><span class="op">(</span>pat<span class="op">=</span><span class="st">'NYYN'</span><span class="op">)</span></span>
<span><span class="va">DNA_seqs</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "ACCA" "CTTC" "GCCG" "TTTT" "ATCA" "CCTC" "GTCG" "TCTT" "ACTA" "CTCC"</span></span>
<span><span class="co">## [11] "GCTG" "TTCT" "ATTA" "CCCC" "GTTG" "TCCT" "ACCC" "CTTG" "GCCT" "TTTA"</span></span>
<span><span class="co">## [21] "ATCC" "CCTG" "GTCT" "TCTA" "ACTC" "CTCG" "GCTT" "TTCA" "ATTC" "CCCG"</span></span>
<span><span class="co">## [31] "GTTT" "TCCA" "ACCG" "CTTT" "GCCA" "TTTC" "ATCG" "CCTT" "GTCA" "TCTC"</span></span>
<span><span class="co">## [41] "ACTG" "CTCT" "GCTA" "TTCC" "ATTG" "CCCT" "GTTA" "TCCC" "ACCT" "CTTA"</span></span>
<span><span class="co">## [51] "GCCC" "TTTG" "ATCT" "CCTA" "GTCC" "TCTG" "ACTT" "CTCA" "GCTC" "TTCG"</span></span>
<span><span class="co">## [61] "ATTT" "CCCA" "GTTC" "TCCG"</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="createdir_delifexists">3.6 createDir_delIfExists<a class="anchor" aria-label="anchor" href="#createdir_delifexists"></a>
</h3>
<div class="section level4">
<h4 id="description-5">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-5"></a>
</h4>
<p>create a directory and delete if it already exists</p>
</div>
<div class="section level4">
<h4 id="usage-5">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-5"></a>
</h4>
<p><code>createDir_delIfExists(dir='testDir')</code></p>
</div>
<div class="section level4">
<h4 id="arguments-5">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-5"></a>
</h4>
<ul>
<li>
<code>dir</code>: Name of the dir to be created</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-5">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-5"></a>
</h4>
<p>create a directory and delete if it already exists</p>
</div>
<div class="section level4">
<h4 id="value-5">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-5"></a>
</h4>
<p>Create a directory of given name. (It will delete the directory if it
is already existing there)</p>
</div>
<div class="section level4">
<h4 id="examples-5">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-5"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/createDir_delIfExists.html">createDir_delIfExists</a></span><span class="op">(</span>dir<span class="op">=</span><span class="st">'testDir'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Old Directory has been deletedNew Directory has been created</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.exists</a></span><span class="op">(</span><span class="st">'testDir'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="context_oligonucscounts">3.7 context_oligonucsCounts<a class="anchor" aria-label="anchor" href="#context_oligonucscounts"></a>
</h3>
<div class="section level4">
<h4 id="description-6">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-6"></a>
</h4>
<p>oligo-nucleotide counts in within a genomic context.</p>
</div>
<div class="section level4">
<h4 id="usage-6">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-6"></a>
</h4>
<p><code>context_oligonucsCounts(contextGr=hg_38_gr, oligoType='trinucs', ignore.strand=FALSE, assmblyName='hg38')</code></p>
</div>
<div class="section level4">
<h4 id="arguments-6">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-6"></a>
</h4>
<ul>
<li>
<code>contextGr</code>: GenomicRange object of the genomic context
within which oligo-nucleotides has to be counted.</li>
<li>
<code>oligoType</code>: dinucs or trinucs or tetranucs</li>
<li>
<code>ignore.strand</code>: genomic context strand information
should be considered. Default: FALSE</li>
<li>
<code>assmblyName</code>: human genome assembly name (hg19 or hg38).
Default: hg19</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-6">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-6"></a>
</h4>
<p>oligo-nucleotide counts in within a genomic context.</p>
</div>
<div class="section level4">
<h4 id="value-6">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-6"></a>
</h4>
<p>a vector of oligo-nucleotide counts</p>
</div>
<div class="section level4">
<h4 id="examples-6">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-6"></a>
</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">hg_38_gr</span><span class="op">)</span></span>
<span></span>
<span><span class="va">oligonucs.Counts</span><span class="op">=</span><span class="fu"><a href="../reference/context_oligonucsCounts.html">context_oligonucsCounts</a></span><span class="op">(</span>contextGr<span class="op">=</span><span class="va">hg_38_gr</span>, oligoType<span class="op">=</span><span class="st">'trinucs'</span>, ignore.strand<span class="op">=</span><span class="cn">FALSE</span>, assmblyName<span class="op">=</span><span class="st">'hg38'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">oligonucs.Counts</span></span></code></pre></div>
<pre><code><span><span class="co">##     AAA     AAC     AAG     AAT     ACA     ACC     ACG     ACT     AGA     AGC </span></span>
<span><span class="co">## 1325123  507680  697059  857418  698085  401071   87610  563602  766871  484544 </span></span>
<span><span class="co">##     AGG     AGT     ATA     ATC     ATG     ATT     CAA     CAC     CAG     CAT </span></span>
<span><span class="co">##  615364  563602  706620  461672  634677  857418  649772  517039  700902  634677 </span></span>
<span><span class="co">##     CCA     CCC     CCG     CCT     CGA     CGC     CGG     CGT     CTA     CTC </span></span>
<span><span class="co">##  630048  447455   94106  615364   76306   81429   94106   87610  451366  581037 </span></span>
<span><span class="co">##     CTG     CTT     GAA     GAC     GAG     GAT     GCA     GCC     GCG     GCT </span></span>
<span><span class="co">##  700902  697059  685228  326166  581037  461672  496828  405020   81429  484544 </span></span>
<span><span class="co">##     GGA     GGC     GGG     GGT     GTA     GTC     GTG     GTT     TAA     TAC </span></span>
<span><span class="co">##  533465  405020  447455  401071  399481  326166  517039  507680  727231  399481 </span></span>
<span><span class="co">##     TAG     TAT     TCA     TCC     TCG     TCT     TGA     TGC     TGG     TGT </span></span>
<span><span class="co">##  451366  706620  677442  533465   76306  766871  677442  496828  630048  698085 </span></span>
<span><span class="co">##     TTA     TTC     TTG     TTT </span></span>
<span><span class="co">##  727231  685228  649772 1325123</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="getpromgrange">3.9 getPromGRange<a class="anchor" aria-label="anchor" href="#getpromgrange"></a>
</h3>
<div class="section level4">
<h4 id="description-7">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-7"></a>
</h4>
<p>Get genomic range of ProteinCodingGenes promoters (upst to -200)</p>
</div>
<div class="section level4">
<h4 id="usage-7">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-7"></a>
</h4>
<p><code>getPromGRange(upst = 500, assmblyName)</code></p>
</div>
<div class="section level4">
<h4 id="arguments-7">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-7"></a>
</h4>
<ul>
<li>
<code>upst</code>: Upstream of the promoter. Promoter region will be
“upst” to -200 to TSS. DEFAULT value is 500</li>
<li>
<code>assmblyName</code>: human genome assembly name (hg19 or hg38).
Default: hg19</li>
</ul>
</div>
<div class="section level4">
<h4 id="value-7">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-7"></a>
</h4>
<p>genomic ranges for protein coding genes and their promoter for a give
assembly.</p>
</div>
<div class="section level4">
<h4 id="examples-7">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-7"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outGr</span><span class="op">=</span><span class="fu"><a href="../reference/getPromGRange.html">getPromGRange</a></span><span class="op">(</span>upst<span class="op">=</span><span class="fl">500</span>, assmblyName<span class="op">=</span><span class="st">'hg19'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned many:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outGr</span><span class="op">$</span><span class="va">proteinCoding.proms</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 18918 ranges and 2 metadata columns:</span></span>
<span><span class="co">##          seqnames            ranges strand |     gene_id</span></span>
<span><span class="co">##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##    79501     chr1       68591-69290      + |       79501</span></span>
<span><span class="co">##   729759     chr1     367159-367858      + |      729759</span></span>
<span><span class="co">##   148398     chr1     860030-860729      + |      148398</span></span>
<span><span class="co">##   339451     chr1     895467-896166      + |      339451</span></span>
<span><span class="co">##    84069     chr1     901377-902076      + |       84069</span></span>
<span><span class="co">##      ...      ...               ...    ... .         ...</span></span>
<span><span class="co">##     1617     chrY 25345055-25345754      - |        1617</span></span>
<span><span class="co">##    57135     chrY 26959440-26960139      - |       57135</span></span>
<span><span class="co">##     9085     chrY 26193962-26194661      - |        9085</span></span>
<span><span class="co">##    57054     chrY 26959440-26960139      - |       57054</span></span>
<span><span class="co">##     9083     chrY 27198052-27198751      - |        9083</span></span>
<span><span class="co">##                                  GenesMap</span></span>
<span><span class="co">##                              &lt;data.frame&gt;</span></span>
<span><span class="co">##    79501   79501:OR4F5:protein-coding:...</span></span>
<span><span class="co">##   729759 729759:OR4F29:protein-coding:...</span></span>
<span><span class="co">##   148398 148398:SAMD11:protein-coding:...</span></span>
<span><span class="co">##   339451 339451:KLHL17:protein-coding:...</span></span>
<span><span class="co">##    84069 84069:PLEKHN1:protein-coding:...</span></span>
<span><span class="co">##      ...                              ...</span></span>
<span><span class="co">##     1617     1617:DAZ1:protein-coding:...</span></span>
<span><span class="co">##    57135    57135:DAZ4:protein-coding:...</span></span>
<span><span class="co">##     9085     9085:CDY1:protein-coding:...</span></span>
<span><span class="co">##    57054    57054:DAZ3:protein-coding:...</span></span>
<span><span class="co">##     9083     9083:BPY2:protein-coding:...</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from hg19 genome</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outGr</span><span class="op">$</span><span class="va">proteinCoding.genes</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 18918 ranges and 2 metadata columns:</span></span>
<span><span class="co">##          seqnames            ranges strand |     gene_id</span></span>
<span><span class="co">##             &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##    79501     chr1       69091-70008      + |       79501</span></span>
<span><span class="co">##   729759     chr1     367659-368597      + |      729759</span></span>
<span><span class="co">##   148398     chr1     860530-879961      + |      148398</span></span>
<span><span class="co">##   339451     chr1     895967-901099      + |      339451</span></span>
<span><span class="co">##    84069     chr1     901877-910484      + |       84069</span></span>
<span><span class="co">##      ...      ...               ...    ... .         ...</span></span>
<span><span class="co">##     1617     chrY 25275502-25345254      - |        1617</span></span>
<span><span class="co">##    57135     chrY 25275502-26959639      - |       57135</span></span>
<span><span class="co">##     9085     chrY 26191377-26194161      - |        9085</span></span>
<span><span class="co">##    57054     chrY 26909216-26959639      - |       57054</span></span>
<span><span class="co">##     9083     chrY 27177050-27198251      - |        9083</span></span>
<span><span class="co">##                                  GenesMap</span></span>
<span><span class="co">##                              &lt;data.frame&gt;</span></span>
<span><span class="co">##    79501   79501:OR4F5:protein-coding:...</span></span>
<span><span class="co">##   729759 729759:OR4F29:protein-coding:...</span></span>
<span><span class="co">##   148398 148398:SAMD11:protein-coding:...</span></span>
<span><span class="co">##   339451 339451:KLHL17:protein-coding:...</span></span>
<span><span class="co">##    84069 84069:PLEKHN1:protein-coding:...</span></span>
<span><span class="co">##      ...                              ...</span></span>
<span><span class="co">##     1617     1617:DAZ1:protein-coding:...</span></span>
<span><span class="co">##    57135    57135:DAZ4:protein-coding:...</span></span>
<span><span class="co">##     9085     9085:CDY1:protein-coding:...</span></span>
<span><span class="co">##    57054    57054:DAZ3:protein-coding:...</span></span>
<span><span class="co">##     9083     9083:BPY2:protein-coding:...</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from hg19 genome</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="gbin_bycgcnts">3.10 Gbin_ByCGcnts<a class="anchor" aria-label="anchor" href="#gbin_bycgcnts"></a>
</h3>
<div class="section level4">
<h4 id="description-8">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-8"></a>
</h4>
<p>Create a GenomicRange of genomic blocks of user specified CpG counts.
(human genome)</p>
</div>
<div class="section level4">
<h4 id="usage-8">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-8"></a>
</h4>
<p><code>Gbin_ByCGcnts(CGs_perBin=100, assmblyName='hg19' )</code></p>
</div>
<div class="section level4">
<h4 id="arguments-8">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-8"></a>
</h4>
<ul>
<li>
<code>assmblyName</code>: hg19 or hg38 or t2t</li>
<li>
<code>CGs_perBin</code>: CpG counts in a genomic block/bin. (even
number) oligo-nucleotides has to be counted.</li>
<li>
<code>addSeq</code>: if sequence of the bin is required (Default:
FALSE)</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-7">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-7"></a>
</h4>
<p>Creates a GenomicRange of genomic blocks of user specified CpG
counts. (human genome)</p>
</div>
<div class="section level4">
<h4 id="value-8">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-8"></a>
</h4>
<p>GenomicRanges object of bins with user specified CpG counts</p>
</div>
<div class="section level4">
<h4 id="examples-8">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-8"></a>
</h4>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hg_19_CpGBins.gr</span><span class="op">=</span><span class="fu"><a href="../reference/Gbin_ByCGcnts.html">Gbin_ByCGcnts</a></span><span class="op">(</span>CGs_perBin<span class="op">=</span><span class="fl">100</span>, assmblyName<span class="op">=</span><span class="st">'hg19'</span> <span class="op">)</span></span>
<span><span class="va">hg_19_CpGBins.gr</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 564328 ranges and 0 metadata columns:</span></span>
<span><span class="co">##            seqnames            ranges strand</span></span>
<span><span class="co">##               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##        [1]     chr1       10469-10761      *</span></span>
<span><span class="co">##        [2]     chr1       10766-11094      *</span></span>
<span><span class="co">##        [3]     chr1       11105-12758      *</span></span>
<span><span class="co">##        [4]     chr1       12773-15190      *</span></span>
<span><span class="co">##        [5]     chr1       15207-17585      *</span></span>
<span><span class="co">##        ...      ...               ...    ...</span></span>
<span><span class="co">##   [564324]     chrY 59355793-59357713      *</span></span>
<span><span class="co">##   [564325]     chrY 59357736-59360397      *</span></span>
<span><span class="co">##   [564326]     chrY 59360409-59361720      *</span></span>
<span><span class="co">##   [564327]     chrY 59361723-59361953      *</span></span>
<span><span class="co">##   [564328]     chrY 59361962-59362400      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 24 sequences from hg19 genome</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="largevariables">3.11 largeVariables<a class="anchor" aria-label="anchor" href="#largevariables"></a>
</h3>
<div class="section level4">
<h4 id="description-9">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-9"></a>
</h4>
<p>Memory usage of large variable in workspace</p>
</div>
<div class="section level4">
<h4 id="usage-9">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-9"></a>
</h4>
<p><code>largeVariables(n)</code></p>
</div>
<div class="section level4">
<h4 id="arguments-9">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-9"></a>
</h4>
<ul>
<li>
<code>n</code>: Number of top memory consuming variables
required</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-8">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-8"></a>
</h4>
<p>Memory of variable in decreasing order</p>
</div>
<div class="section level4">
<h4 id="value-9">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-9"></a>
</h4>
<p>DataFrame of top memory variables</p>
</div>
<div class="section level4">
<h4 id="examples-9">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-9"></a>
</h4>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/largeVariables.html">largeVariables</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                    Class      KB       GB</span></span>
<span><span class="co">## hg_19_Chr.gr     GRanges 3023145 3.023145</span></span>
<span><span class="co">## hg_19_Bins.gr    GRanges   48383 0.048383</span></span>
<span><span class="co">## outGr               list   18654 0.018654</span></span>
<span><span class="co">## hg_19_CpGBins.gr GRanges    4422 0.004422</span></span>
<span><span class="co">## hg_38_gr         GRanges     176 0.000176</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="maketracks_of_grangeslist">3.12 makeTracks_of_grangesList<a class="anchor" aria-label="anchor" href="#maketracks_of_grangeslist"></a>
</h3>
<div class="section level4">
<h4 id="description-10">
<strong>Description</strong><a class="anchor" aria-label="anchor" href="#description-10"></a>
</h4>
<p>make tracks list from GenomicRange lists.</p>
</div>
<div class="section level4">
<h4 id="usage-10">
<strong>Usage</strong><a class="anchor" aria-label="anchor" href="#usage-10"></a>
</h4>
<p><code>makeTracks_of_grangesList(grlist, location, assmblyName)</code></p>
</div>
<div class="section level4">
<h4 id="arguments-10">
<strong>Arguments</strong><a class="anchor" aria-label="anchor" href="#arguments-10"></a>
</h4>
<ul>
<li>
<code>grlist</code>: GenomicRange object of the genomic context
within which oligo-nucleotides has to be counted.</li>
<li>
<code>location</code>: location on the chromosome as List. ie.,
list(chr=‘chr7’,from=26700000, to=26750000)</li>
<li>
<code>assmblyName</code>: human genome assembly name (hg19 or hg38).
Default: hg19</li>
</ul>
</div>
<div class="section level4">
<h4 id="details-9">
<strong>Details</strong><a class="anchor" aria-label="anchor" href="#details-9"></a>
</h4>
<p>Make tracks from genomic ranges objects list</p>
</div>
<div class="section level4">
<h4 id="value-10">
<strong>Value</strong><a class="anchor" aria-label="anchor" href="#value-10"></a>
</h4>
<p>list of tracks for input list of genomic ranges objects.</p>
</div>
<div class="section level4">
<h4 id="examples-10">
<strong>Examples</strong><a class="anchor" aria-label="anchor" href="#examples-10"></a>
</h4>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># example 1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">cpgIslands</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mygrlist</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>gr1<span class="op">=</span><span class="va">cpgIslands</span>,gr2<span class="op">=</span><span class="va">cpgIslands</span><span class="op">)</span></span>
<span><span class="va">loc</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>chr<span class="op">=</span><span class="st">'chr7'</span>,from<span class="op">=</span><span class="fl">26700000</span>, to<span class="op">=</span><span class="fl">26750000</span><span class="op">)</span></span>
<span><span class="va">tracks</span><span class="op">=</span><span class="fu"><a href="../reference/makeTracks_of_grangesList.html">makeTracks_of_grangesList</a></span><span class="op">(</span>grlist<span class="op">=</span><span class="va">mygrlist</span>, location<span class="op">=</span><span class="va">loc</span>, assmblyName<span class="op">=</span><span class="st">'hg19'</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/plotTracks.html" class="external-link">plotTracks</a></span><span class="op">(</span><span class="va">tracks</span>, from <span class="op">=</span> <span class="va">loc</span><span class="op">$</span><span class="va">from</span>, to <span class="op">=</span> <span class="va">loc</span><span class="op">$</span><span class="va">to</span><span class="op">)</span></span></code></pre></div>
<p><img src="CustomBioInfoFunctionsHumanGenome_files/figure-html/unnamed-chunk-14-1.png" width="700"><img src="reference/figures/trackplot1.png"></p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># example 2, only genomic ranges no annotation</span></span>
<span><span class="va">gr1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span><span class="op">)</span>,  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">110</span>,<span class="fl">105</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">120</span>, <span class="fl">150</span> <span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">gr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="st">"chr1"</span>, strand <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"+"</span>, <span class="st">"-"</span>, <span class="st">"+"</span>, <span class="st">"+"</span><span class="op">)</span>,  ranges <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">12</span>,<span class="fl">60</span>, <span class="fl">105</span><span class="op">)</span>, end <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">25</span>, <span class="fl">50</span>, <span class="fl">70</span>, <span class="fl">115</span> <span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Gviz/man/plotTracks.html" class="external-link">plotTracks</a></span><span class="op">(</span><span class="fu"><a href="../reference/makeTracks_of_grangesList.html">makeTracks_of_grangesList</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>gr1<span class="op">=</span><span class="va">gr1</span>,gr2<span class="op">=</span><span class="va">gr2</span><span class="op">)</span>, if_plain<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>, shape<span class="op">=</span><span class="st">"box"</span><span class="op">)</span></span></code></pre></div>
<p><img src="CustomBioInfoFunctionsHumanGenome_files/figure-html/unnamed-chunk-15-1.png" width="700"><img src="reference/figures//trackplot2.png"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Vinod Singh.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
